python/GangaLHCb/Lib/GaudiRunApp/GaudiRunAppHandlers.py

```diff
@@ -0,0 +1,346 @@
+    Args:
+        app (GaudiRun): This expects only the GaudiRun app
+    """
+    opts_file = app.getOptsFile()
+    if isinstance(opts_file, (LocalFile, DiracFile)):
+        # Ideally this would NOT need the basename, however LocalFile is special in this regard.
+        # TODO Fix this after fixing LocalFile
+        opts_name = os.path.basename(opts_file.namePattern)
+    else:
+        raise ApplicationConfigurationError("The filetype: %s is not yet supported for use as an opts file.\nPlease contact the Ganga devs is you wish this implemented." %
+                                            getName(opts_file))
+    full_cmd = "./run gaudirun.py %s %s" % (opts_name, data_file)
+    return full_cmd
+
+class GaudiRunRTHandler(IRuntimeHandler):
+
+    """The runtime handler to run plain executables on the Local backend"""
+
+    def prepare(self, app, appconfig, appmasterconfig, jobmasterconfig):
+        """
+        Prepare the job in order to submit to the Local backend
+        Args:
+            app (GaudiRun): This application is only expected to handle GaudiRun Applications here
+            appconfig (unknown): Output passed from the application configuration call
+            appmasterconfig (unknown): Output passed from the application master_configure call
+            jobmasterconfig (tuple): Output from the master job prepare step
+        """
+
+        job = app.getJobObject()
+
+        prepared_files = []
+
+        # Setup the command which to be run and the input and output
+        input_sand = job.inputsandbox
+        output_sand = job.outputsandbox
+        data_files = genDataFiles(job)
+
+        input_sand = unique(input_sand + prepared_files + data_files)
+
+        job_command = prepareCommand(app)
+
+        # Generate the script which is to be executed for us on the WN
+        scriptToRun = generateWNScript(job_command, job)
+        input_sand.append(scriptToRun)
+
+        # Collect all of the prepared files which we want to pass along to this job
+        input_files, input_folders = collectPreparedFiles(app)
+        if input_folders:
+            raise ApplicationConfigurationError('Prepared folders not supported yet, please fix this in future')
+        else:
+            for f in input_files:
+                input_sand.append(File(f))
+
+        # It's this authors opinion that the script should be in the PATH on the WN
+        # As it stands policy is that is isn't so we have to call it in a relative way, hence "./"
+        c = StandardJobConfig('./'+scriptToRun.name, input_sand, [], output_sand)
+        return c
+
+allHandlers.add('GaudiRun', 'Local', GaudiRunRTHandler)
+#allHandlers.add('GaudiRun', 'Condor', GaudiRunRTHandler)
+allHandlers.add('GaudiRun', 'Interactive', GaudiRunRTHandler)
+allHandlers.add('GaudiRun', 'Batch', GaudiRunRTHandler)
+
+def generateDiracInput(app):
+
+    input_files, input_folders = collectPreparedFiles(app)
+
+    job = app.getJobObject()
+
+    compressed_file = None
```

Comment:
https://github.com/ganga-devs/ganga/pull/578/files#diff-8fb2d96aad0323b4634f00d2b0a026b9R197
https://github.com/ganga-devs/ganga/pull/578/files#diff-8fb2d96aad0323b4634f00d2b0a026b9R209
This is being used isn't it?